### PR TITLE
Upgrade model to ES 6.x

### LINKED ISSUE #99: Upgrade model to ES 6.x
- [x] Index templates use "index_patterns" instead of "template" ES 6.x (#98)
- [x] Option `include_in_all` not supported in ES 6.0.0 and later (#101)
- [x] `_all` field not supported starting in ES 6.x (#46)
- [x] the `_default_` mapping is deprecated in ES 6.x and completely removed in ES 7.x (#111)
- [x] ES index type deprecation in ES 6.x and removal in ES 7.x (#84)
- [x] add tests for backward compatibility with ES 5.x (implemented in eda925b and b31049a)
- [x] bump model version (ac07c1d)

### PR DESCRIPTION
**Pull request by lukas-vlcek**
Closes #99

### CODE REVIEW

**Comment by lukas-vlcek** [comment]
@richm PTAL
I think I am done here. <s>I am still testing this with Elasticsearch 6.x though (unit tests do not run integration with real ES). Given this is larger code update we can start reviewing now.</s>

**Comment by lukas-vlcek** [comment]
Integration tests look ok:
- New index templates can be pushed into ES without any issues
- Newly created indices that matches any of provided `index_patterns` uses mappings defined in the index template
- New indices created using the index templates allow only single index type and that is the `_doc` type. Any documents that are indexed into this index using different type are rejected.

---

Tested using Elasticsearch version[6.8.2], build[oss/tar/b506955/2019-07-24T15:24:41.545295Z], OS[Mac OS X/10.14.6/x86_64], JVM[Oracle Corporation/Java HotSpot(TM) 64-Bit Server VM/1.8.0_202-ea/25.202-b03]

File: namespaces/_default_.yml [hunk 1]
```diff
@@ -1,17 +1,25 @@
-version: 2019.06.24.0
+version: 2019.09.24.0
 
 field_defaults:
   type: keyword
   doc_values: true
   index: true
 
-_default_:
+# NOTE: the key '_index_type_' must be equal to supported.index_type_placeholder value
```
Comment (richm) [inline namespaces/_default_.yml:8]: where is `supported.index_type_placeholder` set?  Is the default value `_index_type_`?

Comment (richm) [reply namespaces/_default_.yml:8]: What happens when we use this file to generate the index templates for ES5?

Comment (lukas-vlcek) [reply namespaces/_default_.yml:8]: > where is `supported.index_type_placeholder` set?

https://github.com/ViaQ/elasticsearch-templates/pull/112/files#diff-f83f12b083f75074fe1aad23efe54c61R20

> Is the default value `_index_type_`?

It is the only possible value and it get replaced depending on which ES version is the target from the following constants:
https://github.com/ViaQ/elasticsearch-templates/pull/112/files#diff-f83f12b083f75074fe1aad23efe54c61R24-R28

> What happens when we use this file to generate the index templates for ES5?

It should be handled correctly. :-)
First, we have tests that compare generated templates against released templates (and we run it for both ES 2.x and ES 5.x released templates now).
Second, you can try to generate templates from within the `templates` folder, run `make` and then navigate to one of the subfolders that are named in the `Makefile` like this:
```
INDEX_PATTERN_DIRS = openshift,collectd_metrics
```

**Comment by richm** [comment]
ok.  lgtm - please squash into a single commit and I think it's ready to be merged

**Comment by lukas-vlcek** [comment]
@richm I am not sure about squashing this time. Most of the commits address single breaking change (ES5 -> ES6). And I would prefer keeping individual commits for historical reasons (should we find there is something broken in the future I could more easily navigate to individual commits and individual files that might be incomplete/broken).

**Comment by richm** [comment]
> I would prefer keeping individual commits for historical reasons

ok - sounds good - then /lgtm

**Pull request merged**